parsers/cxx/cxx_parser_namespace.c

```diff
@@ -54,105 +55,234 @@ boolean cxxParserParseNamespace(void)
 	// namespace;
 
 	int iScopeCount = 0;
-	int iCorkQueueIndex = CORK_NIL;
+	
+	int i;
+
+	int aCorkQueueIndices[MAX_NESTED_NAMESPACES];
+	
+	for(i=0;i<MAX_NESTED_NAMESPACES;i++)
+		aCorkQueueIndices[i] = CORK_NIL;
 
-	for(;;)
+	if(!cxxParserParseNextToken())
 	{
+		// syntax error, but we tolerate this
+		CXX_DEBUG_LEAVE_TEXT("Implicit EOF in cxxParserParseNextToken");
+		return TRUE; // EOF
+	}
+
+	if(cxxTokenTypeIs(g_cxx.pToken,CXXTokenTypeIdentifier))
+	{
+		// OK, check next token to see what's coming after
+
+		CXX_DEBUG_PRINT("Got identifier %s",g_cxx.pToken->pszWord->buffer);
+
+		CXXToken * pFirstIdentifier = g_cxx.pToken;
+		CXXToken * pLastIdentifier = g_cxx.pToken;
+
 		if(!cxxParserParseNextToken())
 		{
 			// syntax error, but we tolerate this
-			CXX_DEBUG_LEAVE_TEXT("Implicit EOF in cxxParserParseNextToken");
+			CXX_DEBUG_LEAVE_TEXT("EOF in cxxParserParseNextToken");
 			return TRUE; // EOF
 		}
-
+	
```

Comment:
This is an example of whitespace, which is not needed.
Before making a PR, please, check your change with git diff. The white spaces are printed with red in git diff output.
See https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration#Formatting-and-Whitespace